src/result.js

```diff
@@ -112,22 +112,34 @@ class Result {
   }
 }
 
-/**
- * `String.replace()` converts '$$' to '$', so we must escape each '$' as '$$';
- * but because we’re using `String.replace()` to do it, we must use '$$$'!
- */
-function escapeForStringReplace(string) {
-  return string.replace(/\$/g, '$$$');
-}
-
 function insertIntoIndexHTML(html, head, body) {
-  html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", escapeForStringReplace(body));
+  if (!html) { return Promise.resolve(html); }
+
+  if (body) {
+    let isBodyReplaced = false;
+    html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", function() {
+      isBodyReplaced = true;
```

Comment:
No just wanted get rid of `escapeForStringReplace` because it was doing extra `.replace` calls, and also come across this comment https://github.com/ember-fastboot/fastboot/issues/76#issuecomment-272971321 so just did two things in one shot :P